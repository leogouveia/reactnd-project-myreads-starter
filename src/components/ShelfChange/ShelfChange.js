import React, { Component } from "react";
import PropTypes from "prop-types";

export class ShelfChange extends Component {
  static propTypes = {
    book: PropTypes.object,
    shelves: PropTypes.array,
    handleChangeShelf: PropTypes.func.isRequired
  };

  constructor(props) {
    super(props);
    this.handleChangeShelf = this.handleChangeShelf.bind(this);
  }

  handleChangeShelf(ev) {
    const { handleChangeShelf, book } = this.props;
    book.oldShelf = book.shelf;
    book.shelf = ev.target.value;
    handleChangeShelf(book);
  }

  render() {
    const { shelves, book } = this.props;
    return (
      <div className="book-shelf-changer">
        <select value={book.shelf} onChange={this.handleChangeShelf}>
          <option value="move" disabled>
            Move to...
          </option>
          {shelves.map((shelf, index) => (
            <option key={`${shelf.id}${index}`} value={shelf.id}>
              {shelf.name}
            </option>
          ))}
        </select>
      </div>
    );
  }
}

export default ShelfChange;
