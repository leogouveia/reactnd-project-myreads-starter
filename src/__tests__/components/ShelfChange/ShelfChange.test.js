import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { ShelfChange } from "../../../components/ShelfChange/ShelfChange";
import { books, shelves } from "../../fixtures/fixtures";

const [book] = books;

describe("<ShelfChange />", () => {
  it("should render book change component", () => {
    const handleChangeShelfMock = jest.fn();
    const wrapper = shallow(
      <ShelfChange
        book={book}
        shelves={shelves}
        handleChangeShelf={handleChangeShelfMock}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
