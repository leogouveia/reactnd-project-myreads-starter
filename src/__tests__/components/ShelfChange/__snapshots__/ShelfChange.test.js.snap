// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShelfChange /> should render book change component 1`] = `
<div
  className="book-shelf-changer"
>
  <select
    onChange={[Function]}
    value="shelf1"
  >
    <option
      disabled={true}
      value="move"
    >
      Move to...
    </option>
    <option
      key="shelf10"
      value="shelf1"
    >
      Shelf 1
    </option>
    <option
      key="shelf21"
      value="shelf2"
    >
      Shelf 2
    </option>
    <option
      key="shelf32"
      value="shelf3"
    >
      Shelf 3
    </option>
  </select>
</div>
`;
